daphne/src/taskprov.rs

```diff
@@ -0,0 +1,171 @@
+use crate::{
+    messages::{
+        taskprov::{QueryConfigVar, TaskConfig, VdafType, VdafTypeVar},
+        Extension, HpkeConfig, Id, ReportMetadata,
+    },
+    vdaf::VdafVerifyKey,
+    DapAbort, DapError, DapQueryConfig, DapTaskConfig, DapVersion, Prio3Config, VdafConfig,
+};
+use prio::codec::Decode;
+use ring::{
+    hkdf::{Prk, Salt, HKDF_SHA256},
+    hmac::HMAC_SHA256,
+    hmac::{sign, Key},
+};
+use std::str;
+use url::Url;
+
+/// SHA-256 of "dap-taskprov-00"
+#[allow(dead_code)]
+pub(crate) const TASK_PROV_SALT: [u8; 32] = [
+    0x4d, 0x63, 0x1a, 0xeb, 0xa8, 0xdf, 0xe0, 0x1b, 0x34, 0x4c, 0x29, 0x2d, 0x17, 0xba, 0x34, 0x9a,
+    0x78, 0x97, 0xbf, 0x64, 0x88, 0x00, 0x55, 0x1c, 0x0d, 0x75, 0x32, 0xab, 0x61, 0x4b, 0xe2, 0x21,
+];
+
+/// Compute the task id of a serialized task config.
+pub fn compute_task_id(serialized: &[u8]) -> Id {
+    let key = Key::new(HMAC_SHA256, &TASK_PROV_SALT);
+    let digest = sign(&key, serialized);
+    let mut b: [u8; 32] = [0; 32];
+    let d = digest.as_ref();
+    b[..32].copy_from_slice(&d[..32]);
+    Id(b)
+}
+
+/// Compute the VDAF pseudorandom key for the pre-shared secret verify_key_init.
+pub(crate) fn compute_vdaf_verify_prk(verify_key_init: &[u8]) -> Prk {
```

Comment:
I would name this in a way that's more indicative of how it's used in the protocol.
```suggestion
/// Extract pseudorandom key from the pre-shared secret used for the "taskprov" extension.
pub(crate) fn extract_prk_from_verify_key_init(verify_key_init: &[u8]) -> Prk {
```